backend/module/eCampLib/src/Types/DateTimeUtc.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace eCamp\Lib\Types;
+
+use DateTimeZone;
+
+class DateTimeUtc extends DateBase {
+    protected string $FORMAT = 'Y-m-d\TH:iP';
+
+    /**
+     * DateTimeUTC constructor.
+     *
+     * @param null|string       $time     Date in string representation
+     * @param null|DateTimeZone $timezone Default timezone is UTC
+     * @param null|string       $format   Allowed format of the datetime
+     *
+     * @throws InvalidDateFormatException
+     * @throws InvalidZoneOffsetException
+     */
+    public function __construct($time = 'now', DateTimeZone $timezone = null, string $format = null) {
+        if (null === $timezone) {
+            $timezone = new DateTimeZone('UTC');
+        } elseif ('UTC' !== $timezone->getName()) {
+            throw new InvalidZoneOffsetException('Invalid zone offset: '.$timezone->getName().'. Should be 0');
+        }
+        if ('now' !== $time) {
+            $this->checkFormat($time, $format);
```

Comment:
I don't really understand why one part of the validations is done in the subclasses and another part is done in a method `checkFormat` of the base class. I propose the following:
* Override the constructor in the base class and do the check there
* Rename the base class to something involving `Utc`, since you are already doing UTC-related checks there anyways
* Remove the duplicate timezone checks from the subclasses DateTimeUtc (line 24) and DateUtc (line 24)
* Add an `abstract function shouldParse(?string $timeString) : boolean;` to the base class, which is then implemented in DateTimeUtc (`return 'now' !== $time;`) and DateUtc (`return 'today' !== $time;`). Use this abstract function in the base class constructor to decide whether you should parse the `$time` and validate the parsed timezone.
* Remove the `checkFormat` method from the base class and the overridden constructors from the subclasses since they are no longer needed.